ok-common-backend/src/main/kotlin/ru/otus/kotlin/messaging/context/MessagingContext.kt

```diff
@@ -0,0 +1,56 @@
+package ru.otus.kotlin.messaging.context
+
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import ru.otus.kotlin.messaging.*
+
+class MessagingContext(
+    var messageIds: List<MessageId> = emptyList(),
+    var channelIds: List<ChannelId> = emptyList(),
+    var channelId: ChannelId = ChannelId.NONE,
+    var profileIdFrom: ProfileId = ProfileId.NONE,
+    var profileIdTo: ProfileId = ProfileId.NONE,
+    var messages: List<Message> = emptyList(),
+    var channels: List<Channel> = emptyList(),
+    var errors: List<Error> = emptyList(),
+    private var _page: Page? = null,
+    private var _channel: Channel? = null,
+    private var _instantMessage: InstantMessage? = null,
+    private var _channelMessage: ChannelMessage? = null,
+    private var _profile: Profile? = null
+) {
+    var page
+        get() = _page ?: throw IllegalArgumentException("page is not set")
+        set(value) {
+            _page = value
+        }
```

Comment:
Для этого не нужно выделять теневую переменную. В Котлине есть backing-fields. Дотсуп к ним через ключевое слово `field`:
```suggestion
var page
        get() = field ?: throw IllegalArgumentException("page is not set")
        set(value) {
            field = value
        }
```

Reply:
Но вообще, контекст предназначен только для транспорта данных через бизнес-цепочку. Логику в него помещать  не следует. И уж тем более он не должен генерировать исключения. Эту логику мы будем делать в модуле бизнес-логики. Здесь она не понадобится и будет только мешать.

Reply:
С одной стороны я согласен, что он для логики не предназначен, с другой - я не согласен с подходом val NONE = MpProposalModel() для пустых моделей. Я считаю, что контекст может быть fail-fast и при отсутствии в нем нужных значений операция должна упасть, нежели пробовать работать с полями пустой модели, что гипотетически может и вообще отработать корректно

Reply:
backing-fields, насколько я понял не решают проблему nullability
теневые переменные имеют класс T?, а видимые переменные уже строго T, в чем и был изначальный смысл)
и могу сказать, что при тестах это весьма помогло - нет значения в контексте - fail-fast